src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -168,15 +180,15 @@ private Entity getLectureEntityFromRequest(HttpServletRequest request) {
   /**
    * Returns URL for the lecture view page for {@code lectureEntity}.
    */
-  private Optional<String> buildRedirectUrl(Entity lectureEntity) {
+  private String buildRedirectUrl(Entity lectureEntity) throws ServletException {
     String lectureId = String.valueOf(lectureEntity.getKey().getId());
-
     try {
-      URIBuilder urlBuilder = new URIBuilder(REDIRECT_URL).addParameter(LectureUtil.ID, lectureId);
-      return Optional.of(urlBuilder.build().toString());
-    } catch (URISyntaxException urlBuilderError) {
-      // TODO: Send a response error.
```

Comment:
This may have belonged in a separate PR, but it greatly simplified how the validation worked in doPost, and took care of a TODO which was closely-related to this PR